sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -226,29 +257,71 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
   if (!approaching_stop_controlled_interction_)
   {
     resp.new_plan.maneuvers = {};
-    ROS_WARN_STREAM("Not approaching stop-controlled itnersection so no maneuvers");
+    ROS_WARN_STREAM("Not approaching stop-controlled intersection so no maneuvers");
     return true;
   }
 
+  bool is_empty_schedule_msg = (scheduled_depart_time_ == 0 && scheduled_enter_time_ == 0 && scheduled_enter_time_ == 0);
+  if (is_empty_schedule_msg)
+  {
+    resp.new_plan.maneuvers = {};
+    ROS_WARN_STREAM("Receiving empty schedule message");
+    return true;
+  }
+
+  ROS_DEBUG("Planning for intersection...");
+
   ROS_DEBUG("Finding car information");
   // Extract vehicle data from request
   VehicleState current_state = extractInitialState(req);
 
   // Get current traffic light information
-  ROS_DEBUG("\n\nFinding intersecction information");
+  ROS_DEBUG("\n\nFinding intersection information");
 
   auto stop_intersection_list = wm_->getIntersectionsAlongRoute({ req.veh_x, req.veh_y });
   auto nearest_stop_intersection = stop_intersection_list.front();
-  double stop_intersection_down_track =
-  wm_->routeTrackPos(nearest_stop_intersection->stopLines().front().front().basicPoint2d()).downtrack;
+  
+  // extract the intersection stop line information
+  std::vector<double> stop_lines;
+
+  for (auto l:nearest_stop_intersection->stopLines())
+  {
+    //TODO: temp use veh_length here until planning stack is updated with front bumper pos
+    double stop_bar_dtd = wm_->routeTrackPos(l.front().basicPoint2d()).downtrack - config_.veh_length;
+    stop_lines.push_back(stop_bar_dtd);
+  }
+  std::sort(stop_lines.begin(), stop_lines.end());
+  
+  double stop_intersection_down_track = stop_lines.front();
+
+  ROS_DEBUG_STREAM("stop_intersection_down_track  " << stop_intersection_down_track);
+  
+  
+  double distance_to_stopline = stop_intersection_down_track - current_downtrack_ - config_.stop_line_buffer;
+  ROS_DEBUG_STREAM("distance_to_stopline  " << distance_to_stopline);
+
+
+  double intersection_end_downtrack = stop_lines.back();
+  // Identify the lanelets of the intersection
+  std::vector<lanelet::ConstLanelet> intersection_lanelets =
+          getLaneletsBetweenWithException(stop_intersection_down_track, intersection_end_downtrack, true, true);
+
+  // find the turn direction at intersection:
+
+  intersection_turn_direction_ = getTurnDirectionatIntersection(intersection_lanelets);
+
+  uint32_t base_time = street_msg_timestamp_;
+  
+  bool time_to_approach_int = (int(scheduled_stop_time_) - int(street_msg_timestamp_))>0;
```

Comment:
I think this whole clause can be brought down a few lines to be:

`if (scheduled_stop_time_ - street_msg_timestamp_ > 0) {...`

Reply:
I think because of the type of these variables, the difference would always come out positive and activate the logic at the wrong time. But if you have a solution for this issue I can make the change.